pkg/strc/middleware_tracer.go

```diff
@@ -0,0 +1,44 @@
+package strc
+
+import (
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+)
+
+func httpRequestWithTracing(r *http.Request) (TraceID, *http.Request) {
+	traceID := TraceIDFromRequest(r)
+	if traceID == EmptyTraceID {
+		traceID = NewTraceID()
+	}
+
+	newCtx := WithTraceID(r.Context(), traceID)
+	r = r.WithContext(newCtx)
+
+	spanID := SpanIDFromRequest(r)
+	if spanID != EmptySpanID {
+		newCtx = WithSpanID(r.Context(), spanID)
+		r = r.WithContext(newCtx)
+	}
+	return traceID, r
+}
+
+func EchoTracer(config MiddlewareConfig) echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			for _, filter := range config.Filters {
+				if !filter(nil, c.Request()) {
+					return next(c)
+				}
+			}
+
+			traceID, req := httpRequestWithTracing(c.Request())
+			c.SetRequest(req)
```

Comment:
Just a note - this is a preemptive extraction of possible common code. We do not know if we need to add stdlib middleware too but if this happens, it is gonna be easy.

So far it looks like:

* crc = echo
* composer cloud api = echo
* composer weldr api = echo
* composer-worker = stdlib

I think we might not even need stdlib middleware.

Reply:
and tbh, we could switch composer-worker to echo.

Reply:
oh composer worker is echo too, only weldr is stdlib, and that's deprecated